app/src/main/java/org/oppia/app/utility/Logger.kt

```diff
@@ -0,0 +1,126 @@
+import android.widget.Toast
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.launch
+import javax.inject.Singleton
+
+/** This Wrapper class is for Android Logs and to perform file logging. */
+@Singleton
+class Logger @Inject constructor(@ApplicationContext context: Context) {
+
+  private val ENABLE_CONSOLE_LOG = true
+  private val ENABLE_FILE_LOG = true
+  private val GLOBAL_LOG_LEVEL = LogLevel.VERBOSE
+  private val LOG_DIRECTORY = File(context.filesDir, "oppia_app.log")
+
+  private enum class LogLevel private constructor(val logLevel: Int) {
+    VERBOSE(Log.VERBOSE),
+    DEBUG(Log.DEBUG),
+    INFO(Log.INFO),
+    WARNING(Log.WARN),
+    ERROR(Log.ERROR),
+    ASSERT(Log.ASSERT)
+  }
+
+  /** Logs a verbose message with the specified tag.*/
+  fun v(tag: String, msg: String) {
+    write(LogLevel.VERBOSE, tag, msg)
+  }
+
+  /** Logs a verbose message with the specified tag, message and exception.*/
+  fun v(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.VERBOSE, tag, msg, tr)
+  }
+
+  /** Logs a debug message with the specified tag*/
+  fun d(tag: String, msg: String) {
+    write(LogLevel.DEBUG, tag, msg)
+  }
+
+  /** Logs a debug message with the specified tag, message and exception.*/
+  fun d(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.DEBUG, tag, msg, tr)
+  }
+
+  /** Logs a info message with the specified tag.*/
+  fun i(tag: String, msg: String) {
+    write(LogLevel.INFO, tag, msg)
+  }
+
+  /** Logs a info message with the specified tag, message and exception.*/
+  fun i(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.INFO, tag, msg, tr)
+  }
+
+  /** Logs a warn message with the specified tag.*/
+  fun w(tag: String, msg: String) {
+    write(LogLevel.WARNING, tag, msg)
+  }
+
+  /** Logs a warn message with the specified tag, message and exception.*/
+  fun w(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.WARNING, tag, msg, tr)
+  }
+
+  /** Logs a error message with the specified tag.*/
+  fun e(tag: String, msg: String) {
+    write(LogLevel.ERROR, tag, msg)
+  }
+
+  /** Logs a error message with the specified tag, message and exception.*/
+  fun e(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.ERROR, tag, msg, tr)
+  }
+
+  private fun isLogEnable(logLevel: LogLevel): Boolean {
+    return GLOBAL_LOG_LEVEL.logLevel < logLevel.logLevel
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String) {
+    if (isLogEnable(logLevel) && ENABLE_CONSOLE_LOG) {
+      Log.println(logLevel.logLevel, tag, log)
+    }
+    if (isLogEnable(logLevel) && ENABLE_FILE_LOG) {
+      val msg = Calendar.getInstance().time.toString() + "\t" + logLevel.name + "/" + tag + ": " + log
+      /** To mange background threads*/
+      CoroutineScope(IO).launch { write(msg) }
+    }
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String, tr: Throwable) {
```

Comment:
I feel like we can consolidate these two write methods into one. What about something like:

```kotlin
private fun writeLog(logLevel: LogLevel, tag: String, log: String) {
  writeInternal(logLevel, tag, log)
}

private fun writeError(logLevel: LogLevel, tag: String, log: String, tr: Throwable) {
  writeInternal(logLevel, tag, "$log\n${Log.getStackTraceString(tr)}")
}

private fun writeInternal(logLevel: LogLevel, tag: String, fullLog: String) {
  // ... Do everything here.
}
```

This helps encourage consistency between the two log formats if they change in the future.

Reply:
made single method to write log